src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java

```diff
@@ -0,0 +1,1899 @@
+                                                     final int exonEnd,
+                                                     final Strand strand) {
+        return createSpliceSiteCodonChange(variantStart, exonNumber, exonStart, exonEnd, strand, 0);
+    }
+
+    /**
+     * Gets a codon change string for a splice site.
+     * Assumes the variant and exon referenced in the params are on the same contig.
+     * @param variantStart Start position (1-based, inclusive) of the variant.
+     * @param exonNumber Number of the exon in the transcript.
+     * @param exonStart Start position (1-based, inclusive) of the exon.
+     * @param exonEnd End position (1-based, inclusive) of the exon.
+     * @param strand The {@link Strand} on which the variant and exon are read.
+     * @param offsetIndelAdjustment An adjustment added to account for bases lost / gained in an Indel event.
+     * @return A {@link String} representing the codon change for the splice site represented by the given parameters.
+     */
+    public static String createSpliceSiteCodonChange(final int variantStart,
+                                                     final int exonNumber,
+                                                     final int exonStart,
+                                                     final int exonEnd,
+                                                     final Strand strand,
+                                                     final int offsetIndelAdjustment) {
+        assertValidStrand( strand );
+
+        char sign = '-';
+        int offset = exonStart - variantStart;
+        if ( Math.abs(offset) > Math.abs(variantStart - exonEnd)) {
+            offset = variantStart - exonEnd;
+            sign = '+';
+        }
+        offset = Math.abs(offset);
+
+        if (strand == Strand.NEGATIVE) {
+            if ( sign == '+' ) {
+                sign = '-';
+            }
+            else {
+                sign = '+';
+            }
+        }
+
+        // Add our indel adjustment here:
+        if ( sign == '+' ) {
+            offset += offsetIndelAdjustment;
+        }
+        else {
+            offset -= offsetIndelAdjustment;
+        }
+
+        // Make sure we correctly adjust for the zero crossing:
+        if ( offset < 0 ) {
+            offset *= -1;
+            if ( sign == '+') {
+                sign = '-';
+            }
+            else {
+                sign = '+';
+            }
+        }
+
+        return "c.e" + exonNumber + sign + offset;
+    }
+
+    /**
+     * Creates the string representation of the codon change for the given {@link SequenceComparison}.
+     * @param seqComp {@link SequenceComparison} representing the alternate and reference alleles for a DNA sequence.
+     * @return A {@link String} representing the codon change for the given {@link SequenceComparison}.
+     */
+    public static String getProteinChangeString(final SequenceComparison seqComp) {
+
+        Utils.nonNull(seqComp);
```

Comment:
Refactored.

Because the `SequenceComparison` can hold the information for any two input sequences, there are cases where the `SequenceComparison` object could be valid and not contain this info (e.g. for an intronic variant).